### PR TITLE
UI Framework: Added 'Attachment' glyph

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Part of the feature request: https://www.ilias.de/docu/goto_docu_wiki_wpage_2276_1357.html

Image of Paperclip

### CODE REVIEW

File: src/UI/Component/Glyph/Factory.php [hunk 1]
```diff
@@ -675,5 +675,28 @@ public function sad($action = null);
 	 */
 	public function angry($action = null);
 
-
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *       The Paperclip Glyph indicates whether or not something is attached to an object or entity.
```
Comment (Amstutz) [inline src/UI/Component/Glyph/Factory.php]: Is the paperclip only shown IF there is an attachment? If yes, then:
**The Paperclip Glyph indicates that something is attached to an object or entity.**

If no, then indicate in effect, that the clip is disabled if there is no attachment (I would however, rather not show at all).

**Review by Amstutz — COMMENTED** [review]
Thanks a lot for adding this!

I have some small uncertainty regarding showing or not showing the clip if there is not attachment at all. If this is cleared, I approve this PR.

**Comment by mjansenDatabay** [comment]
Thx @Amstutz . It should be shown ONLY IF there is an attachment. I just pushed the little change.

**Comment by atoedt** [comment]
The Aria label should not read "paperclip". It should read "attachment"

**Comment by matthiaskunkel** [comment]
Jour Fixe, 09 APR 2018: We highly appreciate this PR. Please change the title to "Attachment glyph" and change the purpose: "The attachment glyph indicates that a file is attached or can be attached to an object or entity."

**Comment by mjansenDatabay** [comment]
Changes according to JF decision applied...

File: src/UI/Component/Glyph/Factory.php [hunk 2]
```diff
@@ -675,5 +675,28 @@ public function sad($action = null);
...
+	 *
+	 * context: Impose the information whether or not files have been attached to emails in the folder view.
+	 *
+	 * rules:
+	 *   composition:
+	 *       1: >
+	 *          A Status Counter MAY indicate the overall amount of attachments.
+	 *   accessibility:
+	 *       1: >
+	 *          The aria-label MUST be 'Attachment'.
```
Comment (klees) [inline src/UI/Component/Glyph/Factory.php:696]: Is this implemented? If not, it seems also no being implemented for the other glyphs, right? Currently I fail to see where in the template the label is rendered.

**Review by klees — APPROVED** [review]
LGTM in general. I would suggest to fix the (potential) aria-label problem in another PR. It also just might be to hot.

**Comment by klees** [comment]
@Amstutz @mjansenDatabay Anything left to do here?

File: src/UI/Component/Glyph/Factory.php (hunk 2 above)
Comment (Amstutz) [reply src/UI/Component/Glyph/Factory.php:696]: @klees : label is passed through constructor in implementation and rendered in line 37 in renderer:

`		$tpl->setVariable("LABEL", $this->txt($component->getAriaLabel()));
`

Note that this is done by using the glyph.context and glyph.standard.html

**Comment by Amstutz** [comment]
@klees Thanks for pumping this up. I answered the open question of yours. @mjansenDatabay plz double checks if this does not introduce any new unit test violations, if not, you can directly merge.

**Comment by Amstutz** [comment]
@mjansenDatabay : There is a little nasty issue that breaks the unit tests (good we have them). However, to wrap this up here, I merge this and fix this immediately afterwards. Thx for the contribution.

**Pull request merged**